src/store/store.ts

```diff
@@ -0,0 +1,1218 @@
+    getCourtOrderStep (): CourtOrderStepIF {
+      return this.stateModel.courtOrderStep
+    },
+
+    getDocumentDelivery (): DocumentDeliveryIF {
+      return this.stateModel.documentDelivery
+    },
+
+    /** The restoration object. */
+    getRestoration (): RestorationStateIF {
+      return this.stateModel.restoration
+    },
+
+    /** The business name validity. */
+    getBusinessNameValid (): boolean {
+      return this.stateModel.restoration.businessNameValid
+    },
+
+    /** The restoration type validity. */
+    getRestorationTypeValid (): boolean {
+      return this.stateModel.restoration.restorationTypeValid
+    },
+
+    /** The approval type validity. */
+    getApprovalTypeValid (): boolean {
+      return this.stateModel.restoration.approvalTypeValid
+    },
+
+    //
+    // Dissolution getters
+    //
+
+    /** The dissolution object. */
+    getDissolution (): DissolutionStateIF {
+      return this.stateModel.dissolution
+    },
+
+    /** The dissolution statement step. */
+    getDissolutionStatementStep (): DissolutionStatementIF {
+      return this.getDissolution.dissolutionStatementStep
+    },
+
+    /** The dissolution type. */
+    getDissolutionType (): DissolutionTypes {
+      return this.getDissolution.dissolutionType
+    },
+
+    getDissolutionHasCertificateDestroyed (): boolean {
+      return this.getDissolution.hasCertificateDestroyed
+    },
+
+    /** Is true when the custodian data is valid. */
+    isDissolutionCustodianValid (): boolean {
+      return this.getDissolution.custodianOfRecords.valid
+    },
+
+    /** The dissolution custodian of records. */
+    getDissolutionCustodian (): OrgPersonIF {
+      return this.getDissolution.custodianOfRecords.custodian
+    },
+
+    /** The custodian email. */
+    getDissolutionCustodianEmail (): string {
+      return this.getDissolutionCustodian?.officer.email
+    },
+
+    /** Is true when the Define Dissolution page is valid. */
+    isDissolutionDefineDissolutionValid (): boolean {
+      const isCoopDefineDissolutionValid = this.isTypeCoop
+        ? (this.getDissolutionStatementStep.valid && this.getDissolutionHasCertificateDestroyed)
+        : true
+
+      return isCoopDefineDissolutionValid && this.isDissolutionCustodianValid
+    },
+
+    /** The registration object. */
+    getRegistration (): RegistrationStateIF {
+      return this.stateModel.registration
+    },
+
+    /** The completing party data. */
+    getCompletingParty (): CompletingPartyIF {
+      return this.stateModel.completingParty
+    },
+
+    /** The dissolution date. */
+    getDissolutionDate (): string {
+      return this.getDissolution.dissolutionDate
+    },
+
+    /** The parties list. */
+    getParties (): Array<PartyIF> {
+      return this.stateModel.parties
+    },
+
+    //
+    // The getters below return values from the current resource
+    // model -- in other words, for the currently-selected entity type.
+    //
```

Comment:
Since the following come from the resource model instead of the state model, should we keep the getters separate?

@JazzarKarim Do you think this should be a different store, or imported separately, or ...?

Reply:
From what I know, different stores are made for different modules. In this case, we're not dealing with that.

Also, Travis mentioned before that we can split the store into multiple files but not sure if that's recommended: https://github.com/vuejs/pinia/issues/802#issuecomment-1018780409

So, I think if we keep it as is, it should be fine. Also, this follows the same path as Edit UI. So, it's consistent. Let me know what you think.

CC: @seeker25

Reply:
I like consistency and I defer to your (new-found 😉) experience and wisdom.

I'm also happy to hear what Travis has to say about this.